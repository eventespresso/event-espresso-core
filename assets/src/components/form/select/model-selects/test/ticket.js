import TicketSelect, { EditorTicketSelect } from '../ticket';
import { shallow, render } from 'enzyme';

jest.mock( '../../model-select', () => () => <span>TicketSelect</span> );

describe( 'TicketSelect()', () => {
	const expectedDefaultQueryData = {
		limit: 100,
		order: 'DESC',
		orderBy: 'start_date',
		showExpired: false,
	};
	it( 'matches snapshot with default props', () => {
		const wrapper = shallow( <TicketSelect /> );
		expect( wrapper ).toMatchSnapshot();
	} );
	it( 'matches expected default props', () => {
		const wrapper = shallow( <TicketSelect /> );
		expect( wrapper.prop( 'queryData' ) )
			.toEqual( expectedDefaultQueryData );
		expect( wrapper.prop( 'selectConfiguration' ).placeholder ).toEqual(
			'Select Ticket...',
		);
		expect( wrapper.prop( 'modelName' ) ).toEqual( 'ticket' );
		expect( wrapper.prop( 'label' ) ).toEqual( 'Select Ticket' );
	} );
	it( 'matches expected update to queryData state with forEventId prop',
		() => {
			const wrapper = shallow( <TicketSelect forEventId={ 20 } /> );
			expect( wrapper.state( 'queryData' ) ).toEqual( {
				...expectedDefaultQueryData,
				forEventId: 20,
			} );
		} );
	it( 'matches expected update to queryData state with forDatetimeId prop',
		() => {
			const wrapper = shallow( <TicketSelect forDatetimeId={ 20 } /> );
			expect( wrapper.state( 'queryData' ) ).toEqual( {
				...expectedDefaultQueryData,
				forDatetimeId: 20,
			} );
		} );
} );

describe( 'EditorTicketSelect()', () => {
	it( 'matches snapshot with default props', () => {
		const wrapper = render( <EditorTicketSelect /> );
		expect( wrapper ).toMatchSnapshot();
	} );
} );
