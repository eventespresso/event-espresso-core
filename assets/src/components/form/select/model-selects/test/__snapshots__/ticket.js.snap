// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorTicketSelect() matches snapshot with default props 1`] = `
<div
  class="components-base-control"
>
  <div
    class="components-base-control__field"
  >
    <span>
      TicketSelect
    </span>
  </div>
</div>
`;

exports[`TicketSelect() matches snapshot with default props 1`] = `
<Component
  getQueryString={[Function]}
  modelName="ticket"
  optionsEntityMap={
    Object {
      "default": Object {
        "label": "TKT_name",
        "value": "TKT_ID",
      },
    }
  }
  queryData={
    Object {
      "limit": 100,
      "order": "DESC",
      "orderBy": "start_date",
      "showExpired": false,
    }
  }
  selectConfiguration={
    Object {
      "defaultValue": undefined,
      "loadingMessage": [Function],
      "noOptionsMessage": [Function],
      "onChange": undefined,
      "placeholder": "Select Ticket...",
    }
  }
  selectLabel="Select Ticket"
/>
`;
